streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/TaskMovementTest.java

```diff
@@ -35,262 +44,161 @@
 
-        assertFalse(
+        assertThat(
             assignTaskMovements(
-                balancedAssignment,
                 emptyMap(),
-                clientStates,
-                getMapWithNumStandbys(allTasks, 1),
-                maxWarmupReplicas)
+                getClientStatesMap(client1, client2, client3),
+                maxWarmupReplicas),
+            is(false)
         );
-        verifyClientStateAssignments(balancedAssignment, emptyWarmupAssignment);
     }
 
     @Test
     public void shouldMoveTasksToCaughtUpClientsAndAssignWarmupReplicasInTheirPlace() {
         final int maxWarmupReplicas = Integer.MAX_VALUE;
-        final Set<TaskId> allTasks = mkSet(TASK_0_0, TASK_0_1, TASK_0_2);
+        final ClientState client1 = getClientStateWithActiveAssignment(singletonList(TASK_0_0));
+        final ClientState client2 = getClientStateWithActiveAssignment(singletonList(TASK_0_1));
+        final ClientState client3 = getClientStateWithActiveAssignment(singletonList(TASK_0_2));
+        final Map<UUID, ClientState> clientStates = getClientStatesMap(client1, client2, client3);
 
-        final Map<UUID, List<TaskId>> balancedAssignment = mkMap(
-            mkEntry(UUID_1, singletonList(TASK_0_0)),
-            mkEntry(UUID_2, singletonList(TASK_0_1)),
-            mkEntry(UUID_3, singletonList(TASK_0_2))
+        final Map<TaskId, SortedSet<UUID>> tasksToCaughtUpClients = mkMap(
+            mkEntry(TASK_0_0, mkSortedSet(UUID_1)),
+            mkEntry(TASK_0_1, mkSortedSet(UUID_3)),
+            mkEntry(TASK_0_2, mkSortedSet(UUID_2))
         );
 
-        final Map<TaskId, SortedSet<UUID>> tasksToCaughtUpClients = new HashMap<>();
-        tasksToCaughtUpClients.put(TASK_0_0, mkSortedSet(UUID_1));
-        tasksToCaughtUpClients.put(TASK_0_1, mkSortedSet(UUID_3));
-        tasksToCaughtUpClients.put(TASK_0_2, mkSortedSet(UUID_2));
-
-        final Map<UUID, List<TaskId>> expectedActiveTaskAssignment = mkMap(
-            mkEntry(UUID_1, singletonList(TASK_0_0)),
-            mkEntry(UUID_2, singletonList(TASK_0_2)),
-            mkEntry(UUID_3, singletonList(TASK_0_1))
-        );
-
-        final Map<UUID, List<TaskId>> expectedWarmupTaskAssignment = mkMap(
-            mkEntry(UUID_1, EMPTY_TASK_LIST),
-            mkEntry(UUID_2, singletonList(TASK_0_1)),
-            mkEntry(UUID_3, singletonList(TASK_0_2))
-        );
-
-        assertTrue(
+        assertThat(
+            "should have assigned movements",
             assignTaskMovements(
-                balancedAssignment,
                 tasksToCaughtUpClients,
                 clientStates,
-                getMapWithNumStandbys(allTasks, 1),
-                maxWarmupReplicas)
-        );
-        verifyClientStateAssignments(expectedActiveTaskAssignment, expectedWarmupTaskAssignment);
-    }
-
-    @Test
-    public void shouldProduceBalancedAndStateConstrainedAssignment() {
```

Comment:
I couldn't follow exactly what this test was doing, but it seemed like other tests already verify both balance and constrainedness, so I deleted this one. Not sure if that's the right call.

Reply:
IIRC this was covering an edge case where it might produce an unbalanced assignment. But it may be moot at this point (and besides, we don't necessarily need to produce a perfectly balanced assignment here)

Reply:
Ah, thanks. I guess I'll just leave it out, then.